src/Compilers/CSharp/Portable/Binder/BinderFactory.BinderFactoryVisitor.cs

```diff
@@ -818,11 +818,11 @@ internal Binder VisitNamespaceDeclaration(NamespaceDeclarationSyntax parent, int
 
             private Binder MakeNamespaceBinder(CSharpSyntaxNode node, NameSyntax name, Binder outer, bool inUsing)
             {
-                QualifiedNameSyntax dotted;
-                while ((dotted = name as QualifiedNameSyntax) != null)
+                if (name is QualifiedNameSyntax dotted)
```

Comment:
Why do can we do away with the `while` here, but in `WithExternAndUsingAliasesBinder.BuildImportChain` we still need to handle multiple dots?

Reply:
> Why do can we do away with the while here, but in WithExternAndUsingAliasesBinder.BuildImportChain we still need to handle multiple dots?

Because they look at different parts of the name. ```WithExternAndUsingAliasesBinder.BuildImportChain``` loops through the lefts. There is nothing to loop through on the right, it is always a simple name.

---
In reply to: [588799216](https://github.com/dotnet/roslyn/pull/51676#discussion_r588799216) [](ancestors = 588799216)